### PR TITLE
Add __serialize/__unserialize

### PR DESCRIPTION
**Pull request by ChristophWurst**
> As of PHP 8.1.0, a class which implements Serializable without also implementing [__serialize()](https://www.php.net/manual/en/language.oop5.magic.php#object.serialize) and [__unserialize()](https://www.php.net/manual/en/language.oop5.magic.php#object.unserialize) will generate a deprecation warning.

@bytestream 

Upstream PR: https://github.com/horde/Stream/pull/1

### CODE REVIEW

**Comment by ChristophWurst** [comment]
The remaining deprecation notices are coming from `horde-stream` and `horde-util`.

Edit: https://github.com/bytestream/Stream/pull/2 and https://github.com/bytestream/Util/pull/4 fix them

**Comment by bytestream** [comment]
Oh no!
```
PHP Deprecated:  strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in /home/runner/work/Mime/Mime/vendor/bytestream/horde-util/lib/Horde/String.php on line 198
```

File: lib/Horde/Mime/Headers/ContentParam.php [hunk 1]
```diff
@@ -403,29 +403,46 @@ public function offsetUnset($offset)
     /**
      */
     public function serialize()
+    {
+        return serialize($this->__serialize());
+    }
+
+    /**
+     */
```
Comment (bytestream) [inline lib/Horde/Mime/Headers/ContentParam.php:411]: okay i like it picasso

**Review by bytestream — APPROVED** [review]

File: lib/Horde/Mime/Headers/ContentParam.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/Horde/Mime/Headers/ContentParam.php:411]: it looks strange in the diff but it is the original docblock of ``unserialize`` :)

**Pull request merged**